### PR TITLE
TW-548 : add notifications bubble to navigation chat icon

### PR DESCRIPTION
**Pull request by imGok**
image

demobadge
demobadgeweb

#548

### CODE REVIEW

**Comment by hoangdat** [comment]
image

It is different in color.

File: lib/widgets/layouts/adaptive_layout/adaptive_scaffold_view.dart [hunk 1]
```diff
@@ -188,15 +191,22 @@ class AppScaffoldView extends StatelessWidget {
   List<NavigationDestination> getNavigationDestinations(BuildContext context) {
     return [
       NavigationDestination(
-        icon: const Icon(Icons.contacts_outlined),
+        icon: const TwakeNavigationIcon(
+          icon: Icons.contacts_outlined,
+          notificationCount: 0,
```
Comment (nqhhdev) [inline lib/widgets/layouts/adaptive_layout/adaptive_scaffold_view.dart:196]: It's default `notificationCount = 0`. Remove it

File: lib/widgets/twake_components/twake_navigation_icon/twake_navigation_icon.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Badge(
+      backgroundColor: Theme.of(context).colorScheme.error,
+      isLabelVisible: notificationCount > 0,
+      largeSize: TwakeNavigationIconStyle.badgeHeight,
+      label: Text(
+        notificationCount.toString(),
+        style: Theme.of(context).textTheme.labelSmall?.copyWith(
+              color: Theme.of(context).colorScheme.onError,
```
Comment (nqhhdev) [inline lib/widgets/twake_components/twake_navigation_icon/twake_navigation_icon.dart:23]: It's same color code `M3/sys light/Error/error` ?

File: lib/widgets/twake_components/twake_navigation_icon/twake_navigation_icon.dart [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            ),
+      ),
+      child: Icon(
+        icon,
+        color: Theme.of(context).colorScheme.onSurfaceVariant,
```
Comment (nqhhdev) [inline lib/widgets/twake_components/twake_navigation_icon/twake_navigation_icon.dart:28]: It's same color code `M3/sys light/BG + Surface/on-surface-variant`?

**Review by drminh2807 — APPROVED** [review]

File: pubspec.yaml [hunk 4]
```diff
@@ -10,7 +10,6 @@ dependencies:
   adaptive_dialog: ^1.8.0+1
   flutter_adaptive_scaffold: ^0.1.4
   animations: ^2.0.7
-  badges: ^2.0.3
```
Comment (nqhhdev) [inline pubspec.yaml:13]: Why don't you use it? Why do you delete it?

File: lib/widgets/layouts/adaptive_layout/adaptive_scaffold_view.dart (hunk 1 above)
Comment (imGok) [reply lib/widgets/layouts/adaptive_layout/adaptive_scaffold_view.dart]: you're right

**Pull request merged**